test/metabase/api/database_test.clj

```diff
@@ -377,43 +408,24 @@
                                                      :substring "a")))))))))))
 
 
-(defn- card-with-native-query {:style/indent 1} [card-name & {:as kvs}]
-  (merge
-   {:name          card-name
-    :database_id   (mt/id)
-    :dataset_query {:database (mt/id)
-                    :type     :native
-                    :native   {:query (format "SELECT * FROM VENUES")}}}
-   kvs))
-
-(defn- card-with-mbql-query {:style/indent 1} [card-name & {:as inner-query-clauses}]
-  {:name          card-name
-   :database_id   (mt/id)
-   :dataset_query {:database (mt/id)
-                   :type     :query
-                   :query    inner-query-clauses}})
-
-(defn- virtual-table-for-card [card & {:as kvs}]
-  (merge
-   {:id               (format "card__%d" (u/the-id card))
-    :db_id            (:database_id card)
-    :display_name     (:name card)
-    :schema           "Everything else"
-    :moderated_status nil
-    :description      nil}
-   kvs))
+(deftest card-autocomplete-suggestions-test
```

Comment:
I think we need more tests for permissions related.
- database perms
- card read perms

Reply:
Done, see https://github.com/metabase/metabase/commit/166708b2315ba382f476bb11fd2c9d1b12b1738b